[pt2e] Avoid getting model device once per node

andrewor14 opened the pull request:
**Summary:** Previously, we call `assert_and_get_unqiue_device` once per node in both prepare and convert. This is expensive and unnecessary since the model device is the same across all nodes, so we should just call this once in the beginning and reuse the same model device across all the nodes.

**Test Plan:**
python test/test_quantization.py -k TestQuantizePT2E


cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

jerryzh168 commented on torch/ao/quantization/utils.py:
```diff
@@ -800,15 +800,6 @@ def _assert_and_get_unique_device(module: torch.nn.Module) -> Any:
-            "Both 'meta' and 'cpu' are present in the list of devices. Module can have one device. We Select 'cpu'."
-        )
-        devices = {torch.device("cpu")}
-    ""
```
let me check

iremyux commented:
Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI and its test purposes - nothing about this PR specifically. (It should not effect the acceptance of the PR even if it fails.)

andrewor14 committed:
[pt2e] Avoid getting model device once per node

**Summary:** Previously, we call `assert_and_get_unqiue_device`
once per node in both prepare and convert. This is expensive and
unnecessary since the model device is the same across all nodes,
so we should just call this once in the beginning and reuse the
same model device across all the nodes.

**Test Plan:**
python test/test_quantization.py -k TestQuantizePT2E

ghstack-source-id: 3e72b14591df5e2b06871a76c8e0bf4108366ed3
Pull Request resolved: https://github.com/pytorch/pytorch/pull/162012

jerryzh168 commented on torch/ao/quantization/observer.py:
```diff
@@ -1902,10 +1902,18 @@ def convert(self, model: torch.fx.GraphModule, observer_node: Node):
+                    model.graph,
+                    "_scale",
+                    scale,
+                    scale.device,
```
value could be a non-Tensor I think? according to `create_getattr_from_value` L274, so should this be the device from model?
